src/grid.js

```diff
@@ -15,18 +22,28 @@ export type Props = {
   justify?: Justify,
   margin?: Spacing,
   padding?: Spacing,
+  marginTop?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  paddingTop?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
   size?: Size,
 }
 
+/* eslint-disable react/prefer-stateless-function */
 export default class Grid extends React.Component {
-  static defaultProps = {
-    margin: [],
-  }
-
   static contextTypes = {
     devMode: PropTypes.bool,
   }
 
+  componentWillReceiveProps(newProps: Props) {
+    validateSpacingProps(newProps, 'margin')
+    validateSpacingProps(newProps, 'padding')
```

Comment:
Oh I see now you are applying the validation separately. I think we should do this as part of setting the spacing classes. I don't think it's the responsibility of this component to set everything and then check it was the right thing